proxy/src/test/java/com/velocitypowered/proxy/scheduler/VelocitySchedulerTest.java

```diff
@@ -65,4 +66,39 @@ void repeatTaskWorks() throws Exception {
     task.cancel();
   }
 
+  @Test
+  void nullSchedulerTest() throws Exception {
+    VelocityScheduler scheduler = new VelocityScheduler(new FakePluginManager());
+
+    assertThrows(NullPointerException.class, () -> scheduler.builder(null).schedule());
+    assertThrows(IllegalArgumentException.class, () -> scheduler.builder(FakePluginManager.PLUGIN_A)
+        .schedule());
+  }
+
+  @Test
+  void obtainTasksFromPlugin() throws Exception {
+    VelocityScheduler scheduler = new VelocityScheduler(new FakePluginManager());
+
+    scheduler.builder(FakePluginManager.PLUGIN_A)
+      .task(ScheduledTask::cancel)
+      .delay(100, TimeUnit.MILLISECONDS)
```

Comment:
This test is time-dependent, and could fail when running with a single core. Could we perform the assertion inside the task itself (and so the task no longer needs to be delayed)?

Reply:
With this specific test I doubt very much that it could fail, since I doubt that going from one simple method to another would take 100 milliseconds